test/integration/belongsToMany.test.js

```diff
@@ -314,6 +334,114 @@ describe('belongsToMany', function () {
     });
   });
 
+  setups.forEach(([description, setup]) => {
```

Comment:
`belongsToMany` was already caching `paranoid` on options, we just needed to add some tests.